wayland-server/src/lib.rs

```diff
@@ -3,67 +3,73 @@
 //! ## Overview
 //!
 //! This crate provides the interfaces and machinery to safely create servers
-//! for the wayland protocol. It is a rust wrapper around the `libwayland-server.so`
+//! for the Wayland protocol. It is a rust wrapper around the `libwayland-server.so`
 //! C library.
 //!
-//! The wayland protocol revolves around the creation of various objects and the exchange
+//! The Wayland protocol revolves around the creation of various objects and the exchange
 //! of messages associated to these objects. Whenever a client connects, a `Display` object
 //! is automatically created in their object space, which they use as a root to create new
 //! objects and bootstrap their state.
 //!
 //! ## Protocol and messages handling model
 //!
-//! The protocol being bi-directional, you can send and receive messages. Sending messages is
-//! done via methods of `Resource<_>` objects, receiving and handling them is done by providing
-//! implementations.
+//! The protocol being bi-directional, you can send and receive messages.
+//! Sending messages is done via methods of Rust objects corresponding to the wayland protocol
+//! objects, receiving and handling them is done by providing implementations.
 //!
 //! ### Resources
 //!
 //! The protocol and message model is very similar to the one of `wayland-client`, with the
-//! main difference being that the handles to objects are represented by the `Resource<I>`
-//! type.
-//!
-//! These resources are used to send messages to the clients (they are called "events" in the
-//! wayland context). This is done by the `Resource::<I>::send(..)` method.
-//!
-//! There is not a 1 to 1 mapping between `Resource<I>` instances and protocol objects. Rather,
-//! you can think of `Resource<I>` as an `Rc`-like handle to a wayland object. Multiple instances
-//! of it can exist referring to the same protocol object.
-//!
-//! Similarly, the lifetimes of the protocol objects and the `Resource<I>` are not tightly tied.
-//! As protocol objects are created and destroyed by protocol messages, it can happen that an object
-//! gets destroyed while one or more `Resource<I>` still refers to it. In such case, these resources
-//! will be disabled and their `alive()` method will start to return `false`. Events that are
-//! subsequently sent to them are ignored.
+//! main difference being that the underlying handles to objects are represented by the `Resource<I>`
+//! type, very similarly to proxies in `wayland-client`.
+//!
+//! These resources are used to send messages to the client (in the Wayland context,
+//! these are called "events"). You usually don't use them directly, and instead call
+//! methods on the Rust objects themselves, which invoke the appropriate `Resource` methods.
+//! It is also possible to directly use the `Resource::<I>::send(..)` method, but
+//! this should only be done carefully: using it improperly can mess the protocol
+//! state and cause protocol errors.
```

Comment:
Server side, the methods are really just shorthands for the `send(..)` method, and provide no additional safety. The same care should be used for both.